lib/modules/manager/cpanfile/extract.ts

```diff
@@ -0,0 +1,16 @@
+import type { PackageFileContent } from '../types';
+import { parse } from './parser';
+
+export function extractPackageFile(
+  content: string,
+  packageFile?: string
+): PackageFileContent | null {
+  const result = parse(content);
+  if (!result?.deps.length) {
+    return null;
+  }
+
+  const { deps, perlVersion } = result;
+  const extractedConstraints = perlVersion ? { perl: perlVersion } : undefined;
+  return { deps, extractedConstraints };
```

Comment:
```suggestion
  return { deps, ...(extractedConstraints && { extractedConstraints }) };
```
Please don't add an undefined extractedConstraints property to result